pymilvus/orm/collection.py

```diff
@@ -430,15 +431,21 @@ def insert(self, data: Union[List, pandas.DataFrame], partition_name: str = None
             >>> res.insert_count
             10
         """
-        check_insert_or_upsert_data_schema(self._schema, data)
-        entities = Prepare.prepare_insert_or_upsert_data(data, self._schema)
+        if data is None:
+            return MutationResult(data)
 
+        row_based = check_insert_or_upsert_is_row_based(data)
         conn = self._get_connection()
-        res = conn.batch_insert(self._name, entities, partition_name,
-                                timeout=timeout, schema=self._schema_dict, **kwargs)
-
-        if kwargs.get("_async", False):
-            return MutationFuture(res)
+        if not row_based:
+            check_insert_or_upsert_data_schema(self._schema, data)
+            entities = Prepare.prepare_insert_or_upsert_data(data, self._schema)
+            res = conn.batch_insert(self._name, entities, partition_name,
+                                    timeout=timeout, schema=self._schema_dict, **kwargs)
+            if kwargs.get("_async", False):
+                return MutationFuture(res)
+        else:
+            res = conn.insert_rows(self._name, data, partition_name,
```

Comment:
Wouldn't it make sense to do the conversion that is found in Prepare. row_insert_or_upsert_param() here and instead of having both insert_batch and insert_rows()?

Reply:
Yes, initially I had contemplated such an approach. However, upon closer inspection, I noticed that even after converting the data into the format required for batch_insert, batch_insert would still proceed to perform another conversion to generate the insertRequest. In light of performance considerations, I therefore opted to provide an insert_rows function which directly transforms into insertRequest.

Reply:
In the future, if we decide to abandon batch_insert, it might be better to provide a unified entry point for inserts in milvus_client that can handle both row and column format.